modules/transport-netty4/src/test/java/org/elasticsearch/http/netty4/Netty4BadRequestTests.java

```diff
@@ -87,7 +88,8 @@ public void dispatchBadRequest(RestChannel channel, ThreadContext threadContext,
             }
         };
 
-        try (HttpServerTransport httpServerTransport = new Netty4HttpServerTransport(Settings.EMPTY, networkService, bigArrays, threadPool,
+        Settings settings = Settings.builder().put(HttpTransportSettings.SETTING_HTTP_PORT.getKey(), getPortRange()).build();
```

Comment:
NIT: maybe extract `Settings.builder().put(HttpTransportSettings.SETTING_HTTP_PORT.getKey(), getPortRange()).build()` to a method somewhere in the test framework module to dry this up a little?